src/train_cmd.cpp

```diff
@@ -2454,16 +2458,23 @@ class VehicleOrderSaver {
 		old_dest_tile(_v->dest_tile),
 		old_last_station_visited(_v->last_station_visited),
 		index(_v->cur_real_order_index),
-		suppress_implicit_orders(HasBit(_v->gv_flags, GVF_SUPPRESS_IMPLICIT_ORDERS))
+		suppress_implicit_orders(HasBit(_v->gv_flags, GVF_SUPPRESS_IMPLICIT_ORDERS)),
+		restored(false)
 	{
 	}
 
-	~VehicleOrderSaver()
+	void restore()
```

Comment:
function should start with a capital letter (we're not the STL!), and missing documentation.

Though I'm not quite clear why it can't just be part of the destructor (documentation!)

Reply:
I fixed the case and documented the function.

I restore the order a bit before returning from ChooseTrainTrack. Doing it RAII-style with the destructor of VehicleOrderSaver would introduce a new 100-line scope in an already complex function.
I added Restore() to make this more readable.